lib/Dialect/Arith/Conversions/ArithToModArith/ArithToModArith.cpp

```diff
@@ -0,0 +1,122 @@
+#include "lib/Dialect/Arith/Conversions/ArithToModArith/ArithToModArith.h"
+
+#include "lib/Dialect/ModArith/IR/ModArithAttributes.h"
+#include "lib/Dialect/ModArith/IR/ModArithOps.h"
+#include "lib/Dialect/ModArith/IR/ModArithTypes.h"
+#include "lib/Utils/ConversionUtils/ConversionUtils.h"
+#include "llvm/include/llvm/Support/Debug.h"            // from @llvm-project
+#include "mlir/include/mlir/IR/BuiltinAttributes.h"     // from @llvm-project
+#include "mlir/include/mlir/IR/ImplicitLocOpBuilder.h"  // from @llvm-project
+#include "mlir/include/mlir/IR/MLIRContext.h"           // from @llvm-project
+#include "mlir/include/mlir/IR/PatternMatch.h"          // from @llvm-project
+#include "mlir/include/mlir/IR/TypeUtilities.h"         // from @llvm-project
+#include "mlir/include/mlir/Support/LLVM.h"             // from @llvm-project
+#include "mlir/include/mlir/Transforms/DialectConversion.h"  // from @llvm-project
+#include "mlir/include/mlir/Transforms/GreedyPatternRewriteDriver.h"  // from @llvm-project
+
+#define DEBUG_TYPE "arith-to-mod-arith"
+
+namespace mlir {
+namespace heir {
+namespace arith {
+
+#define GEN_PASS_DEF_ARITHTOMODARITH
+#include "lib/Dialect/Arith/Conversions/ArithToModArith/ArithToModArith.h.inc"
+
+static mod_arith::ModArithType convertArithType(Type type) {
+  auto modulus = type.getIntOrFloatBitWidth();
```

Comment:
This modulus seems incorrect. An `i32` changes to `mod_arith.int<32 : i32>`? I think it should instead be a modulus of `2**31` (1 + max value of signed int32).

Also note that mod_arith has limitations on the allowed modulus size vs the container type. So if you wanted it to be `2**32` then you'd need to have the container type be at least `i33` or larger.

Reply:
Woops, is indeed quite a big fault. Thanks for catching!